src/components/ContentListTable/ContentListTable.tsx

```diff
@@ -51,15 +46,43 @@ const ContentListTable = () => {
   const [page, setPage] = useState(1);
   const [perPage, setPerPage] = useState(storedPerPage);
 
+  const {
+    debouncedSearchQuery,
+    debouncedSelectedVersions,
+    debouncedSelectedArches,
+    repoParams
+  } = useContext(ContentListContext)
+
   const hasActionPermissions = true; // TODO: Incorporate permissions here later.
 
+  const getLabels = (type: string, names: Array<string>) => {
+    const namesLabels = type === 'arch' ? repoParams['distribution_arches'] : repoParams['distribution_versions'];
+    const labels: Array<string> = [];
+    names.forEach(name => {
+          const found = namesLabels.find(v => v.name === name )
+          if(found) {
+            labels.push(found.label)
+          }
+        }
+    );
+    return labels
+  };
+
```

Comment:
Could this be done in context? As the labels are consumed here, the logic repoParams and versions/arches can take place in context as well.

Reply:
I'm a little unclear on what you mean. Are you saying to move this function to context and just call it in ContentListTable?

Reply:
No, I mean as you only consume these variables here: 
debouncedSearchQuery,
    debouncedSelectedVersions,
    debouncedSelectedArches, 
 
 .. I would suggest placing your function inside context, and have the "Labels" used in this component. 
 So the labels would be passed down by context.

Reply:
good idea! thanks